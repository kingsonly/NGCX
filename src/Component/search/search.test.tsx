/**
 * @jest-environment jsdom
 */

import "@testing-library/jest-dom";

import { fireEvent, render, screen, waitFor, } from "@testing-library/react";

import React from "react";
import { SearchInput } from ".";
import userEvent from "@testing-library/user-event";


function inputEl(inputEl: any, arg1: string) {
  throw new Error("Function not implemented.");
}

describe("Should Render Search Component", () => {
  test("renders search correctly", () => {
    const component = render(<SearchInput />);
    expect(component).toMatchSnapshot();
  });

  test("test Input values", async () => {
    render(<SearchInput />);

    const inputEl = screen.getByTestId("search-input");
    userEvent.type(inputEl, "test@mail.com");
 
    expect(screen.getByTestId("search-input")).toHaveValue("test@mail.com");

  });
});


