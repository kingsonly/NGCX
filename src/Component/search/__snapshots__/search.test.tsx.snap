// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Search Component renders search correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-between w-full"
      >
        <input
          class="outline-none px-2  w-full border-l placeholder:text-xs placeholder:text-gray-300"
          data-testid="search-input"
          placeholder="Search for items"
        />
        <span
          class="px-4"
        >
          <svg
            color="lightgrey"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            style="color: lightgrey;"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-between w-full"
    >
      <input
        class="outline-none px-2  w-full border-l placeholder:text-xs placeholder:text-gray-300"
        data-testid="search-input"
        placeholder="Search for items"
      />
      <span
        class="px-4"
      >
        <svg
          color="lightgrey"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: lightgrey;"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
